core-web/libs/portlets/edit-ema/portlet/src/lib/services/dot-page-api.service.ts

```diff
@@ -79,7 +85,17 @@ export class DotPageApiService {
         const url = params.url.replace(/^\/+|\/+$/g, '');
 
         const pageType = params.clientHost ? 'json' : 'render';
-        const apiUrl = `/api/v1/page/${pageType}/${url}?language_id=${params.language_id}&com.dotmarketing.persona.id=${params['com.dotmarketing.persona.id']}&mode=EDIT_MODE`;
+
+        const pageApiUrl = createUrlWithQueryParams(url, {
+            language_id: params.language_id ?? '1',
+            'com.dotmarketing.persona.id':
+                params['com.dotmarketing.persona.id'] ?? DEFAULT_PERSONA.identifier,
+            variantName: params.variantName ?? DEFAULT_VARIANT_ID,
+            mode: EDIT_MODE,
+            experimentId: params.experimentId
+        });
```

Comment:
Why don't you let the `createUrlWithQueryParams` handle all the fallbacks?

Reply:
Now it handles fallbacks